db.tf

```diff
@@ -89,63 +89,26 @@ resource "aws_rds_cluster_instance" "apiary_cluster_instance" {
-  depends_on = [aws_rds_cluster_instance.apiary_cluster_instance]
-
-  provisioner "local-exec" {
-    command = "${path.module}/scripts/db-iam-auth.sh &> /dev/null"
-
-    environment = {
-      MYSQL_HOST            = "${aws_rds_cluster.apiary_cluster[0].endpoint}"
-      MYSQL_MASTER_USER     = "${aws_rds_cluster.apiary_cluster[0].master_username}"
-      MYSQL_MASTER_PASSWORD = "${aws_rds_cluster.apiary_cluster[0].master_password}"
-    }
-  }
+resource "random_string" "secret_seed_slug" {
+  count   = "${var.external_database_host == "" ? var.db_instance_count : 0}"
+  length  = 8
+  special = false
 }
 
-resource "null_resource" "mysql_rw_user" {
-  count      = "${var.external_database_host == "" ? 1 : 0}"
-  depends_on = [aws_rds_cluster_instance.apiary_cluster_instance]
-
-  triggers = {
-    secret_string = "${md5(data.aws_secretsmanager_secret_version.db_rw_user.secret_string)}"
-  }
-
-  provisioner "local-exec" {
-    command = "${path.module}/scripts/mysql-user.sh &> /dev/null"
-
-    environment = {
-      MYSQL_HOST            = "${aws_rds_cluster.apiary_cluster[0].endpoint}"
-      MYSQL_MASTER_USER     = "${aws_rds_cluster.apiary_cluster[0].master_username}"
-      MYSQL_MASTER_PASSWORD = "${aws_rds_cluster.apiary_cluster[0].master_password}"
-      MYSQL_DB              = "${var.apiary_database_name}"
-      MYSQL_SECRET_ARN      = "${data.aws_secretsmanager_secret.db_rw_user.arn}"
-      MYSQL_PERMISSIONS     = "ALL"
-      AWS_REGION            = "${var.aws_region}"
-    }
-  }
+resource "aws_secretsmanager_secret" "apiary_mysql_master_credentials" {
+  count                   = "${var.external_database_host == "" ? var.db_instance_count : 0}"
+  name                    = "apiary_db_master_user_${random_string.secret_seed_slug.result}"
+  tags                    = var.apiary_tags
+  recovery_window_in_days = 0
 }
 
-resource "null_resource" "mysql_ro_user" {
-  count      = "${var.external_database_host == "" ? 1 : 0}"
-  depends_on = [aws_rds_cluster_instance.apiary_cluster_instance]
-
-  triggers = {
-    secret_string = "${md5(data.aws_secretsmanager_secret_version.db_ro_user.secret_string)}"
-  }
-
-  provisioner "local-exec" {
-    command = "${path.module}/scripts/mysql-user.sh &> /dev/null"
-
-    environment = {
-      MYSQL_HOST            = "${aws_rds_cluster.apiary_cluster[0].endpoint}"
-      MYSQL_MASTER_USER     = "${aws_rds_cluster.apiary_cluster[0].master_username}"
-      MYSQL_MASTER_PASSWORD = "${aws_rds_cluster.apiary_cluster[0].master_password}"
-      MYSQL_DB              = "${var.apiary_database_name}"
-      MYSQL_SECRET_ARN      = "${data.aws_secretsmanager_secret.db_ro_user.arn}"
-      MYSQL_PERMISSIONS     = "SELECT"
-      AWS_REGION            = "${var.aws_region}"
-    }
-  }
-}
+resource "aws_secretsmanager_secret_version" "apiary_mysql_master_credentials" {
+  count         = "${var.external_database_host == "" ? var.db_instance_count : 0}"
```

Comment:
If this is only created if `var.external_database_host == ""`, then is seems like lower down, the logic of launching the init container needs to change if `var.external_database_host == ""`.

I agree, this all becomes a bit nebulous.  We need someone with more context on how external database host works to give feedback.

Reply:
@rpoluri @massdosage could either of you provide more context on how this should be handled properly? (additionally run through a review of this PR when you have a few minutes 😅 )

Reply:
Sorry for the long delay in reply, this dropped off the edge of my TODO list. I'm afraid I don't know the answer, hopefully @rpoluri can comment?

Reply:
with external_database_host, assumption is we won't be managing DB and credentials, so we shouldn't be running the init container also.